### PR TITLE
feat(orchestration): re-assert the orchestrator remit on compaction and /clear

### LINKED ISSUE #787: Orchestrator remit is never re-asserted after /clear — only after compaction, and never on reconnect
## Problem

The orchestrator's role prompt (the pointer telling an orchestration tab to read its generated context file and adopt the orchestrator role/delegation protocol) is currently re-delivered by exactly two triggers:

1. **Spawn time** — when the orchestration tab/pane is first created.
2. **Auto-compaction** — a fix re-asserts the remit when the start-role pane's status transitions to `Compacting`, derived from Claude Code's native `PreCompact` hook.

Reconnect deliberately never replays the prompt either (an explicit design decision, not an oversight).

**`/clear` is a third case nothing currently covers.** When a user runs `/clear` inside an orchestrator pane, Claude Code fires its native `SessionStart` hook with a `source` field of `"clear"` (the same hook also fires with `source: "startup"`, `"resume"`, and `"compact"`). Today the hook payload's `source` field isn't captured or read at all — every `SessionStart` is treated identically regardless of origin, and the re-arm logic only ever watches for a `Compacting` status, never a `SessionStart` event itself.

Net effect: after `/clear`, the orchestrator pane has a genuinely fresh context with no memory of its role, available agents, or delegation protocol — worst exactly when a clean slate makes re-seeding that context most valuable — and nothing re-injects it. The user has to manually re-run the "read your context file" instruction themselves.

## Proposed fix

Extend the existing compaction re-arm mechanism to also fire on a `SessionStart` hook event whose `source` is `"clear"`:

- Capture the hook payload's `source` field (currently dropped) and forward it narrowly into the internal event's metadata for `SessionStart` events only.
- Treat "this pane just reported a clear-originated SessionStart" as an edge trigger for the render loop, alongside the existing `Compacting`-status edge trigger.
- Reuse the existing delivery/retry/confirmation machinery unchanged — this is a second re-arm trigger, not a new delivery path.
- Scope to Claude Code initially, matching the existing compaction re-arm's own scope (other supported agents don't share this hook shape).

## Also filed on the fork

A matching issue is open at https://github.com/prageethw/dot-agent-deck/issues/655, where the fix will land first; a PR will be offered here once it's merged there.

### PR DESCRIPTION
**Pull request by prageethw**
## Summary

The orchestrator's start-role prompt (its "remit") is delivered once at spawn and never re-asserted. Role adherence decays across a long, compacted session — worst exactly when the roles matter most. This adds two independent, edge-triggered re-arm paths, both reusing the existing spawn-time delivery machinery (`deliver_orchestrator_prompt`) unchanged — never a second delivery path:

1. **Compaction re-assertion**: when the start-role pane's own session is first observed `Compacting` (the daemon status the `PreCompact` hook drives), the render loop re-prepares the role prompt and re-enters the same readiness-gated, `SendResult`-confirmed delivery path used at spawn.
2. **`/clear` re-assertion**: Claude Code's native `SessionStart` hook carries a `source` field (`"startup"`/`"resume"`/`"compact"`/`"clear"`). `ClaudeCodeHookInput` now decodes it (Claude-Code-only, lenient-string decoding so a non-string shape degrades to `None` instead of dropping the whole event) and `build_event_typed` forwards only the `"clear"` case into `AgentEvent.metadata` under a new narrow key (`session_start_source=clear`), mirroring the existing `SESSION_START_ORIGIN_METADATA_KEY` forwarding pattern. A second, independent render-loop edge trigger watches for this metadata on the start-role pane's `recent_events` and re-arms the same delivery path.

Both triggers exclude Pi start roles (delivered natively, daemon-side — not via TUI-owned PTY keystroke injection) and any tab whose prompt was already permanently abandoned (deadline or terminal `SendResult`), and neither overwrites a terminal role status (`Done`/`Failed`) on re-delivery.

No wire/protocol change: `AgentEvent.metadata` already flows to the attached TUI's own `AppState` verbatim via the existing broadcast + `apply_event` path, so the new key rides that unchanged — no `PROTOCOL_VERSION` bump.

Closes #787.

## Why

Role prompts aren't just style guidance here — they're the only thing enforcing hard boundaries like "the orchestrator coordinates, it never implements, reviews, or audits itself; anything touching source code or tests must be delegated." That boundary is stated exactly once, in the `prompt_template` sent at spawn, and nothing re-asserts it afterward ([#423](https://github.com/vfarcic/dot-agent-deck/issues/423)).

In practice it decays exactly when it matters most. A real orchestrator session, several hours in and past a compaction, whose spawn prompt said in as many words *"You NEVER do implementation, review, or audit work yourself"* went ahead and wrote a test file and its catalog entries itself instead of delegating that step. Nothing was ambiguous — the instruction forbade it outright — what failed was durability: the seed prompt had aged out of context while other re-injected material (the project's own `CLAUDE.md`) was still fully present. There's no obvious point on the "close an issue → edit a PRD → write a changelog → write a test file" gradient where behavior visibly flips, so nothing short of putting the remit back in context stops it.

Patching this per-project (restating the boundary somewhere the harness re-injects every turn) works, but it's a workaround for something the framework itself causes: dot-agent-deck defines `[[orchestrations.roles]]` and owns remit delivery, so every project using it inherits the same decay and would have to independently rediscover and repatch it. It also gets worse in the direction the product is heading — longer autonomous runs and scheduled dispatch mean more elapsed context between the seed prompt and the moment a role has to decide whether some action is its job.

The fix re-asserts the remit **in the product** instead, reusing machinery dot-agent-deck already has — the daemon already injects prompts into panes; this is the same delivery path spawn uses, not a new one — triggered on the two events that actually wipe out a session's context: auto-compaction and `/clear`. (`/clear` specifically was flagged separately in [#787](https://github.com/vfarcic/dot-agent-deck/issues/787) as the one context-resetting event the original compaction fix didn't yet cover.)

## Background

This bundles two pieces of work from a downstream fork (`prageethw/dot-agent-deck`) that were never previously offered upstream:

- The compaction-re-assertion mechanism (issue #423's fix on that fork), offered here for the first time.
- The `/clear` extension (fork PR [prageethw/dot-agent-deck#656](https://github.com/prageethw/dot-agent-deck/pull/656), already merged there), which builds directly on the compaction mechanism and cannot stand alone without it.

They're bundled into one PR rather than sequenced because #2 has a hard dependency on #1's render-loop scaffolding (`orchestration_prompt_anchor_at`, `orchestration_remit_abandoned`, the Pi-exclusion policy, `next_start_role_status_after_delivery`) that doesn't exist on `main` yet.

This PR ports the **current, combined** production-code behavior as it exists on the fork today — a fresh port against upstream's current file shapes, not a replay of the fork's own commit history.

**Note on #787's framing**: that issue was written from the fork's own vantage point, where the compaction mechanism already existed. On this repo's `main`, compaction re-assertion doesn't exist yet either — this PR introduces it, not just the `/clear` half. Also, the compaction trigger is deliberately agent-agnostic (it keys on `SessionStatus::Compacting`, set by any producer's `EventType::Compacting`), while only the `/clear` trigger is Claude-Code-scoped — #787's text implies both share the same scope, which isn't quite accurate. Leaving a comment on #787 with this correction for the record.

## A bug this port found and fixed in itself

The initial production-code port (`22ca2c42`) silently dropped two pieces from the compaction re-arm loop that the `/clear` loop (ported correctly) has: the `no_delivery_pending` eligibility check that treats a landed-but-unconfirmed delivery as re-arm-eligible (not just a fully-absent one), and three delivery-state resets (`send_retry_backoff`, `prompt_delivery`, `orchestration_ready_since`) that discard a stale prior delivery cycle before starting a new one. This passed `fmt`/`clippy`/`linkage-check` cleanly but failed a genuine e2e test deterministically — 3/3 reruns — with the pane's status line reading `Orchestrator prompt not delivered (the agent's conversation changed); abandoned`. Root cause and fix are in `eae8f013`; the short version is that a still-unconfirmed spawn delivery was left stale, its bound session id went out of date once the pane's generation advanced, and the daemon's own delivery-target-changed guard correctly refused to write into what it read as a different conversation — the fix makes the compaction loop discard that stale delivery before re-arming, matching the `/clear` loop's own behavior.

**This is why the test evidence below matters more than usual for this PR**: static checks alone did not catch this regression; only actually running the tests did.

## Known, deliberate behavior worth your attention (not fixed in this PR)

- **The two re-arm loops are inconsistent on transient-block behavior**, inherited unchanged from the fork: the compaction loop consumes its edge even when a re-arm attempt is blocked by a transient condition (a delivery still probing before its first write, or `prepare_orchestrator_prompt` returning `None`), silently dropping that compaction episode's re-assertion; the `/clear` loop deliberately leaves the edge unmarked so a later frame retries. This is the exact mechanism the bug above rode in on — the fix narrowed the window rather than closing it. Not changed here since it would be an unreviewed behavior change to logic this PR is otherwise faithfully porting; flagging so it's a deliberate choice, not a discovery, if you want a follow-up to align the two loops.
- **A compaction/`/clear` re-assertion always delivers the no-task prompt variant** (`prepare_orchestrator_prompt(config, cwd, None)`), even for an orchestration started via `dispatch --task` or the per-issue dispatch path. A long dispatched run that compacts gets told to "wait for instructions" rather than continue, and the `## Your task` section is dropped from the rewritten context file. This is identical behavior on the fork (not a port defect) and was out of scope for the original fork PRD too — flagging because your dispatch path is active and you'd rather hear this from the PR than the field.
- `prepare_orchestrator_prompt`'s write to `.dot-agent-deck/orchestrator-context.md` has no symlink guard — pre-existing, unrelated to this diff, content is non-secret and the actor is already same-uid; now writes repeat more often over a long session than before.

## Test plan

All six ported tests plus all new unit tests are green, both in upstream CI (fast tier — this repo's CI runs no e2e/L2 tier at all, by deliberate design, for any PR) and in an authorized local run of the e2e tier (the only way to get runtime confirmation of the L2 tests here):

- [x] `orchestration_remit_001_start_role_compaction_reasserts_remit`
- [x] `orchestration_remit_002_non_start_role_compaction_reasserts_nothing`
- [x] `orchestration_remit_003_reassertion_waits_for_confirmed_delivery`
- [x] `orchestration_remit_004_start_role_clear_reasserts_remit`
- [x] `orchestration_remit_005_non_start_role_clear_reasserts_nothing`
- [x] `orchestration_remit_006_non_claude_agent_type_clear_reasserts_nothing`
- [x] `source_001_non_string_source_does_not_drop_the_event`, `clear_session_start_source_forwards_narrowly`, `remit_reassert_fires_only_for_compacting_status`, `remit_pane_compacting_ignores_placeholder_and_other_panes`, `remit_pane_latest_clear_session_start_filters_and_picks_latest`, `next_start_role_status_preserves_terminal_statuses` (all fast-tier, CI-confirmed)

Local e2e run (`cargo nextest run -p dot-agent-deck --features e2e -E 'test(orchestration_remit_)' --no-fail-fast`, after the `eae8f013` fix):
```
Summary [ ... ] 6 tests run: 6 passed, 9228 skipped
```
`orchestration_remit_001` (the test that caught the bug above) was re-run 3/3 times individually to rule out a timing flake before concluding it was a genuine defect, and again after the fix to confirm the fix, also 3/3.

**A documented timing margin**: `MAX_PAYLOAD_SUBMISSIONS` is `2` here (vs `1` on the fork), so a spawn-time delivery unconfirmed 500ms after its write earns one automatic replacement payload write. The ported tests' log-count assertions are only stable while the fixture's `confirm_submission` round-trip completes inside that 500ms window — normally comfortable, but undocumented, so a note explaining this was added to `ORCHESTRATOR_REMIT_SCRIPT`'s doc comment (`tests/e2e_orchestration_remit.rs`) so a future flake investigation starts there instead of from scratch.

**Environment note**: `cargo clippy --workspace --all-targets --features e2e -- -D warnings` fails in this environment because `desktop/src-tauri`'s `libdbus-sys` build script needs `pkg-config`/`libdbus-1-dev`, not installed here — unrelated to this change (no `Cargo.toml` edits). `cargo clippy -p dot-agent-deck --all-targets --features e2e -- -D warnings` (the crate this PR actually touches) is clean, and CI's own `--workspace --all-targets` run confirms clean in its environment.

### CODE REVIEW

**Review by vfarcic — CHANGES_REQUESTED** [review]
Verified locally at the `origin/main` merge (`1e5d2d9`), including the e2e tier and a cross-version run. Two things need to change before this goes in; everything else is filed as follow-ups and does not block.

## What checks out

- **All gates green.** fmt, clippy (full rule-2 command — `dot-agent-deck-desktop` compiled and linted clean here), `build --release`, linkage-check, windows-cross. Fast tier 3443/3443. The e2e tier had 4 failures, all of which pass in isolation — parallel-run contention, including `orchestration_remit_003`, which failed at its *precondition* with both role panes showing "No agent". Not defects.
- **The port is faithful.** Since the fork is public I diffed against `prageethw/dot-agent-deck#656` (`fb70aa1`) rather than taking the restore commits on trust: all four new helpers, both re-arm loops, the `source` field, `lenient_string` and the forwarding block are byte-identical once comments are stripped, and **the test assertions are identical too — no bound widened or lengthened**. `37eaf05`'s restored clause matches the fork word for word.
- **The tests genuinely fail without the fix.** Reverting `src/ui.rs` to `origin/main` gives 5 failures out of 6.
- **`eae8f01`'s ordering claim holds.** The re-arm drops `prompt_delivery` before setting `orchestrator_prompt`, and the new cycle mints a fresh `delivery_id` and reads a fresh watermark before its first write, so an old in-flight write cannot confirm the new one.
- **Rule 12 cross-version test run, and it passes.** v0.39.0 daemon → v0.39.0 TUI to bring the roles up → close it → this branch's TUI, mismatch prompt naming both live agents, declined. One `Attach protocol listening` line and the same daemon pid throughout. Compaction re-delivered; `session_start_source=clear` **also** re-delivered through the old daemon; `delegate --to worker` routed end to end. **So `.feature` rather than `.breaking` is right** — the #243 contrast holds, because there an old daemon actively misbehaved on the new stream, whereas here the daemon is a pass-through for one key on an already-free-form map.

## Required 1 — a re-assertion destroys a dispatched orchestration's task

Both re-arm sites call `prepare_orchestrator_prompt(config, cwd, None)`. That unconditionally rewrites `.dot-agent-deck/orchestrator-context.md` with no `## Your task`, and delivers the no-task pointer ("Acknowledge your role and **wait for instructions**") instead of "Then carry out that task".

Measured, not inferred: I appended `## Your task` plus a sentinel to a live context file, injected one `Compacting` on the start-role pane, and the sentinel was gone — 0 matches — while the pane received the wait-for-instructions line.

So a `dot-agent-deck dispatch --task` orchestration (`dispatch.rs:403` → `spawn.rs:821`) that compacts has its task deleted from disk and is told to stop — on exactly the long-running dispatched path this feature exists to serve. Before this PR that file was written once at spawn and survived compaction, so it is a regression rather than a pre-existing gap, and it silently contradicts `docs/develop/dispatcher-mode.md:39`.

You flag this in the PR body, and I'd rather have it disclosed there than find it in the field — but it shouldn't ship. Two small fixes, your pick:

- carry the task on `Tab::Orchestration` (populated on the dispatch and reattach paths) and pass `Some(task)` at both sites; or
- have the re-arm read the existing file's `## Your task` section back before rewriting — non-destructive, no new tab state.

## Required 2 — the Greptile P2 needs an answer

It landed at 08:46:53Z, after your last commit at 08:30:02Z, so nothing on the branch responds to it — and `required_review_thread_resolution` means the PR cannot merge while it is open.

I checked the close path myself: the finding is correct — `close_tab_by_index` (`ui.rs:9111`) removes only pane-keyed state and nothing removes a `TabId`-keyed entry. **But it is a pre-existing pattern, not something you introduced** — `main` already leaks three collections the same way, and your own doc comments say so. This PR takes it from 3 to 8, tens of bytes per closed tab. It is section 1 of #797, so answering the thread with that and resolving it is all this needs.

## Also worth fixing here

**The changelog fragment overstates coverage.** It scopes the `/clear` trigger to Claude Code but describes the compaction half unconditionally. In practice `EventType::Compacting` only ever comes from `PreCompact`, installed for Claude Code and Codex only — Devin installs `PostCompaction` (mapped to `Thinking`), OpenCode has no compaction event at all, and Pi is excluded by design. One sentence naming real coverage would stop an OpenCode user reading it as covering them. The coverage gap itself is section 3 of #797; it is the wording I would fix in this PR.

## One question

`PreCompact` fires *before* compaction, so the re-assertion types the pointer into a pane that is about to be compacted. Claude Code also emits `SessionStart` with `source: "compact"` *after*, when the context is definitively gone. I would expect the confirm-and-retry machinery to absorb the early write, and this is a port of something that reportedly works on the fork — but my e2e evidence used a synthetic `Compacting` injection, which does not model real compaction timing. Did you see the real ordering behave on the fork?

Full local report and the cross-version transcript are on my side if any of it is useful.

File: src/ui.rs [hunk 1]
```diff
@@ -2357,8 +2459,90 @@ struct UiState {
...
+    /// (success or permanent exclusion), so a later `/clear` isn't
+    /// gratuitously throttled by a stale entry — but that cleanup only
+    /// runs while `latest_clear` is still `Some`. If the triggering event
+    /// ages out of `recent_events` (or is filtered by the tab anchor)
+    /// while a retry entry is outstanding, `latest_clear` goes `None`, the
+    /// whole gate stops running for that tab, and the entry is simply left
+    /// behind. Harmless: the next edge's floor check compares against that
+    /// long-past `Instant` and passes immediately, so the stale entry never
+    /// throttles anything — it just isn't removed.
+    orchestration_remit_clear_retry_at: HashMap<TabId, std::time::Instant>,
```
[the remark this answers is not in the document]
Comment (prageethw) [reply src/ui.rs:2523]: Confirmed and correct — traced `close_tab_by_index` (`src/ui.rs:9111`) myself. But this is pre-existing, not something this PR introduces: `orchestration_prompted`, `orchestration_created_at`/`orchestration_prompt_anchor_at` and `orchestration_ready_since` already leak the same way on `main`. This PR takes it from 3 `TabId`-keyed collections to 8 (adding `orchestration_remit_compacting`, `_clear_reasserted_at`, `_clear_tab_anchor_at`, `_clear_retry_at`, `_abandoned`) — tens of bytes per closed orchestration tab, not a leak anyone will notice in practice.

Filed as section 1 of #797 (https://github.com/vfarcic/dot-agent-deck/issues/797), which groups this with the other non-blocking follow-ups from this review. Resolving here per that issue's own note: "This is the Greptile P2 on #789. It predates that PR — resolving the thread by pointing here is enough."

**Comment by prageethw** [comment]
Addressed both required items from the review, pushed at `4a42c33f`.

**Required 1 (dispatched task destroyed on re-assertion)** — went with option (b): `reassert_orchestrator_prompt` (`src/orchestrator_context.rs`) now reads the existing context file's `## Your task` section back off disk before rewriting it, and carries it forward. Both `src/ui.rs` re-arm sites now call it instead of `prepare_orchestrator_prompt(config, cwd, None)` directly. No new `Tab::Orchestration` state — the file the daemon already wrote has the task. Pinned by 3 unit tests in `orchestrator_context.rs` plus a new L2 case, `orchestration_remit_007`, which seeds a task the way `spawn.rs`'s dispatch path does and confirms both the task-carrying pointer variant and the task itself survive a compaction re-assertion. Verified red against the pre-fix code (stashed the fix, reran — fails as expected, "Orchestrator prompt not delivered … abandoned"), green after (3/3 individual reruns).

**Required 2 (unanswered Greptile P2)** — replied on the thread and resolved it, pointing at #797 section 1 per that issue's own note ("resolving the thread by pointing here is enough").

**Also fixed**: the changelog fragment's compaction claim, scoped to what actually fires it — Claude Code and Codex only (Devin installs `PostCompaction`→`Thinking` instead, OpenCode has no compaction event, Pi is native-delivery). Matches the table already in #797 section 3.

**Gates**: `fmt --check` clean; `clippy -p dot-agent-deck --all-targets --features e2e -D warnings` clean (workspace-wide clippy still blocked here by the pre-existing missing-pkg-config gap your own review's environment note flagged — unrelated to this diff); `linkage-check: ok`; fast tier 3446/3446 (up 3 from your 3443 baseline — the new unit tests); the 7-test `orchestration_remit_` e2e suite green including the new `007`.

Left your one open question (real `PreCompact`-before vs `SessionStart(compact)`-after timing on the fork) for you to weigh in on — I don't have independent field evidence to add there.

**Review by vfarcic — APPROVED** [review]
Approving after an independent delta verification of `37eaf05..4a42c33` (the two commits pushed since the last review).

**Note on the range:** the previously reviewed `1e5d2d9` was a merge commit that has since been dropped — the head descends from `37eaf05`, the merge-base. So `git diff 1e5d2d9..4a42c33` shows two `issue-labeler*.lock.yml` files that are *not* authored work. The real delta is 5 files, +218/-5. Gates were therefore re-run both at the head and on a local merge with `main`.

**Required 1 — verified independently, not carried over.** Reverting *only* the two `src/ui.rs` call sites back to `prepare_orchestrator_prompt(config, cwd, None)` turns the suite red at exactly one test — `orchestration_remit_007` — on the correct assertion, with the grid printing the defect verbatim (the no-task "wait for instructions" pointer delivered twice, task gone). Restoring goes 7/7 green. Note `remit_007` is the *only* test guarding the wiring; the three new unit tests call the helper directly and stay green on a ui.rs-only regression.

Parser edge cases probed with throwaway tests, all clean: two re-assertions are byte-identically idempotent (`## Your task` count stays 1); a task body containing `## Your task` and `###` headings survives verbatim across three passes; empty section, absent section, and absent file all degrade correctly to the no-task pointer; a multiline task's last line survives. This holds structurally because the task section is always appended last and nothing in the context body contains the marker.

**Clippy gap closed.** `cargo clippy --workspace --all-targets --features e2e,e2e-live -- -D warnings` exits 0 on the merged tree with `dot-agent-deck-desktop` linted. The blocker was #780 (`pkg-config` + `tauri-deps` + `PKG_CONFIG_PATH`), which is not in this branch — that is exactly the missing pkg-config. Also worth knowing: `--features e2e,e2e-live` cannot run on this branch as-is, since `e2e-live` does not exist until #502 is merged in.

Gates: fmt clean; fast tier 3481/3481 at head and 3649/3649 merged; `orchestration_remit_` 7/7 both; linkage-check ok (11 rules).

**Changelog scoping verified against the code**, not against #797's table: `hook.rs:144` maps `PreCompact` alone to `EventType::Compacting`. Claude Code and Codex install it; Devin installs `PostCompaction` -> `Thinking` with `PreCompact` explicitly excluded; OpenCode has no compaction event; Pi is excluded in `ui.rs`. The fragment matches exactly.

**One open question, named rather than left implicit.** The compaction timing assumption is still unverified by anyone, and two halves of it are now settled from the code — one unfavourably:

- There is **no post-compaction safety net.** `hook.rs` forwards the `SessionStart` `source` field only when it equals `"clear"` ("every other `source` value stays dropped"), and the `/clear` re-arm requires that metadata key. So `SessionStart{source:"compact"}` sets status to `Idle` and re-arms the edge for a *future* compaction, but triggers no re-delivery for the current one.
- The write is **not gated on compaction finishing** — readiness is only `agent_ready`, true mid-compaction — so attempt 1 types into the pane ~500ms after the `PreCompact` edge.

If that write is lost: two payload writes total (`MAX_PAYLOAD_SUBMISSIONS = 2`), then submit-only probes, then at 60s `abandon_orchestrator_prompt` inserts into `orchestration_remit_abandoned`, which is insert-only and permanent by design. So the downside is not one skipped re-assertion — the tab becomes permanently ineligible for every future remit re-assertion. That severity is the argument for measuring it.

It cannot be settled in the harness today: `inject_compacting` posts a synthetic `AgentEvent` onto the hook socket with no real context loss, and no test anywhere drives a real `/compact`. Settling it needs a lane-2 (`e2e-live`) test with a real Claude Code agent and a real `/compact`, asserting `log_prompt_confirmed` fires for the re-assertion. This is a property of the feature as a whole, not a regression introduced by this delta — worth a tracked follow-up, not a merge block.

Greptile's P2 read and agreed: valid, pre-existing, correctly deferred to #797.

`desktop-web` is red on an unrelated time-based pnpm `minimumReleaseAge` rejection of `lucide-react@1.38.0`; it is failing on `main` at `edaa2b4` too, this PR touches no JS or lockfile, and it is not a required check.

Non-blocking nits for a follow-up: the writer at `orchestrator_context.rs:197` uses a bare literal rather than `TASK_SECTION_MARKER`, so the two could drift (the round-trip unit test would catch it); and the "task section must remain last" invariant is load-bearing for the parser but unstated.
